src/net/http.rs

```diff
@@ -1404,18 +1406,7 @@ impl HttpRequestType {
             return Err(net_error::DeserializeError("Invalid Http request: expected non-zero-length body for PostTransaction".to_string()));
         }
 
-        // content-type must be given, and must be application/octet-stream
-        match preamble.content_type {
-            None => {
-                return Err(net_error::DeserializeError("Missing Content-Type for transaction".to_string()));
-            },
-            Some(ref c) => {
-                if *c != HttpContentType::Bytes {
-                    return Err(net_error::DeserializeError("Wrong Content-Type for transaction; expected application/octet-stream".to_string()));
-                }
-            }
-        };
-
+        // assume application/octet-stream
```

Comment:
Why not enforce the content-type?

Reply:
Reverted in 2849cc4.  Had thought that the content-type wouldn't matter, but turns out, it does in curl (which will send url-encoded data by default).